### PR TITLE
feat: deploy weka cluster with host project

### PR DESCRIPTION
**Pull request by denise89p**

### CODE REVIEW

File: modules/network/variables.tf [hunk 1]
```diff
@@ -81,19 +81,6 @@ variable "allow_weka_api_cidrs" {
   default     = []
 }
 
-
-variable "vpc_connector_region_map" {
```
Comment (assafgi) [inline modules/network/variables.tf:85]: Why did you remove it from here?

File: cloud_functions.tf [hunk 2]
```diff
@@ -12,6 +11,32 @@ locals {
   # common function for multiple actions
   cloud_internal_function_name = "${var.prefix}-${var.cluster_name}-weka-functions"
   function_ingress_settings    = var.subnet_autocreate_as_private ? "ALLOW_INTERNAL_ONLY" : "ALLOW_ALL"
+  vpc_connector_id             = var.vpc_connector_id == "" ? google_vpc_access_connector.connector[0].id : var.vpc_connector_id
+  project_number               = data.google_project.project.number
```
Comment (assafgi) [inline cloud_functions.tf:15]: lets call it `deployment_project_number` , basically I would like also the variable name to be `deployment_project_id`, but lets leave it for now.

Comment (denise89p) [reply cloud_functions.tf:15]: changed

File: README.md [hunk 3]
```diff
@@ -130,11 +136,14 @@ To join an SMB cluster in Active Directory, need to run manually command:
 
 
 ## Shared project
-
+To enable using Shared VPC, with project as a host project and attach other service project.
```
Comment (assafgi) [inline README.md:139]: To setup shared VPC network:

File: health_check.tf [hunk 4]
```diff
@@ -1,6 +1,7 @@
 # health check
 resource "google_compute_region_health_check" "health_check" {
   name                = "${var.prefix}-${var.cluster_name}-health-check"
+  project             = var.project_id
```
Comment (assafgi) [inline health_check.tf:4]: why do we need it? it can be different then the provider project id?

File: health_check.tf [hunk 5]
```diff
@@ -12,23 +13,25 @@ resource "google_compute_region_health_check" "health_check" {
   }
 }
 
-
 # backend service
 resource "google_compute_region_backend_service" "backend_service" {
   name                  = "${var.prefix}-${var.cluster_name}-lb-backend"
+  project               = var.project_id
```
Comment (assafgi) [inline health_check.tf:19]: why do we need it? it can be different then the provider project id?

File: health_check.tf [hunk 6]
```diff
@@ -12,23 +13,25 @@ resource "google_compute_region_health_check" "health_check" {
...
     group = google_compute_instance_group.this.self_link
   }
-  depends_on = [module.network, google_compute_instance_group.this]
+  depends_on = [module.network, module.shared_vpc_peering, google_compute_instance_group.this]
 }
 
 # forwarding rule
 resource "google_compute_forwarding_rule" "google_compute_forwarding_rule" {
   name                  = "${var.prefix}-${var.cluster_name}-forwarding-rule"
+  project               = var.project_id
```
Comment (assafgi) [inline health_check.tf:34]: why do we need it? it can be different then the provider project id?
